src/typedetection.jl

```diff
@@ -99,8 +99,8 @@ function detect(types, io, positions, parsinglayers, kwargs, typemap, categorica
         pools = Vector{CategoricalPool{String, UInt32, CatStr}}(undef, cols)
         for col = 1:cols
             T = types[col]
-            if length(levels[col]) / sum(values(levels[col])) < .67 && T !== Missing && Base.nonmissingtype(T) <: String
-                types[col] = substitute(T, CategoricalArrays.catvaluetype(Base.nonmissingtype(T), UInt32))
+            if T === String || T === Union{String, Missing}
```

Comment:
This makes me worried though for String `id` columns; sure I want a low cardinality string column to be made Categorical, but for a highly unique column, does it really make sense?

Reply:
Not really, but people should think about what they do if they set `categorical=true`. We could always allow passing a number to choose a threshold, but I prefer providing simple options first. We've had reports by users who were confused by this clever logic.

It would also make sense to be able to combine `categorical=true` with e.g. `types=(:ID=>String)` to choose to keep `:ID` as a string column.

Reply:
See https://github.com/JuliaData/CSV.jl/pull/317.